marklogic-data-hub-central/src/test/java/com/marklogic/hub/central/controllers/FlowMvcTest.java

```diff
@@ -0,0 +1,35 @@
+package com.marklogic.hub.central.controllers;
+
+import com.marklogic.hub.central.AbstractMvcTest;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+import org.junit.jupiter.api.Test;
+import org.springframework.mock.web.MockMultipartFile;
+import org.springframework.test.web.servlet.request.MockMvcRequestBuilders;
+import org.springframework.test.web.servlet.result.MockMvcResultHandlers;
+
+import javax.ws.rs.core.MediaType;
+import java.nio.charset.StandardCharsets;
+
+
+public class FlowMvcTest extends AbstractMvcTest {
+
+    private final static String PATH = "/api/flows";
+
+    @Test
+    void runIngestionStep() throws Exception {
+        installReferenceModelProject();
+        setTestUserRoles("hub-central-user","data-hub-operator");
+        loginAsTestUser();
+
+        MockMultipartFile file1 = new MockMultipartFile("file1", "file1.json", MediaType.APPLICATION_JSON, "{\"name\": \"Joe\"}".getBytes(StandardCharsets.UTF_8));
+        MockMultipartFile file2 = new MockMultipartFile("file2","file2.json", MediaType.APPLICATION_JSON, "{\"name\": \"John\"}".getBytes(StandardCharsets.UTF_8));
+
+        mockMvc.perform(MockMvcRequestBuilders.multipart(PATH + "/{flowName}/steps/{stepNumber}", "ingestToFinal", "1")
+                        .file(file1)
+                        .file(file2)
+                        .contentType(MediaType.MULTIPART_FORM_DATA)
+                        .session(mockHttpSession))
+                .andDo(MockMvcResultHandlers.log())
+                .andExpect(status().isOk());
```

Comment:
I think we need to verify that the two JSON documents were ingested as well. This is good in that it verifies no error occurred, but it should be a simple matter to verify that the two documents are in the staging database too - I imagine you can assert on the size of the collection that the documents were ingested into.

Reply:
runFlow() starts the job and returns , so had to include a function to check for completion of job and verified if the docs were ingested.